sdk/monitor/azure-opentelemetry-exporter-azuremonitor/azure/opentelemetry/exporter/azuremonitor/export/_base.py

```diff
@@ -148,12 +154,13 @@ def is_retryable_code(response_code: int) -> bool:
     Determine if response is retryable
     """
     return bool(response_code in (
-        206,  # Retriable
+        206,  # Partial success
         408,  # Timeout
         429,  # Throttle, too Many Requests
         439,  # Quota, too Many Requests over extended time
         500,  # Internal Server Error
         503,  # Service Unavailable
+        504,  # Gateway timeout
```

Comment:
Would be good to understand all retriable codes, .NET have a few more like 502

https://github.com/microsoft/ApplicationInsights-dotnet/blob/develop/BASE/src/ServerTelemetryChannel/Implementation/ResponseStatusCodes.cs